File: Programmers/[64064] 튜플/n__aj22/64065.py
def solution(s):
    answer = []
    #{{}} 제거
    s = s.replace("{", " ") #왼쪽 괄호 공백으로 치환
    s = s.replace("}", " ") #오른쪽 괄호 공백으로 치환
    s = s.lstrip() #왼쪽 공백 제거
    s = s.rstrip() #오른쪽 공백 제거
    s = s.split(" , ") #공백,공백 기준 split
    s.sort(key = lambda x:len(x))

    for value in s:
        value_list = value.split(",")
        for num in value_list:
            num = int(num)
            if num not in answer:
                answer.append(num)
                break
    return answer

if __name__ == "__main__":
    s = ["{{2},{2,1},{2,1,3},{2,1,3,4}}",
    "{{1,2,3},{2,1},{1,2,4,3},{2}}",
    "{{20,111},{111}}",
    "{{123}}",
    "{{4,2,3},{3},{2,3,4,1},{2,3}}"]
    for i in s:
        print(solution(i))

File: Programmers/[64065] 튜플/lululalyang/_64065.java
import java.io.*;
import java.util.*;
public class _64065 {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		String s = br.readLine();
		
		int[] res = solution(s);
		for(Integer i : res)	System.out.print(i + " ");
	}

	
	private static int[] solution(String s) {
		int[] answer = {};
		s = s.substring(2, s.length()-2);
		String[] set = s.split("},");
		int N = set.length;
		answer = new int[N];
		for(int i=1; i<N; i++) {
			set[i] = set[i].substring(1, set[i].length());
		}
		
		Arrays.sort(set, new Comparator<String>() {
			@Override
			public int compare(String s1, String s2) {
				return s1.length() - s2.length();
			}
		});
		
		Map<Integer, Boolean> chk = new HashMap<>();
		int idx = 0;
		for(String str : set) {
			boolean flag = false;
			String[] S = str.split(",");
			int len = S.length;
			for(int i=0; i<S.length; i++) {
				int now = Integer.parseInt(S[i]);
				if(!chk.containsKey(now)) {
					chk.put(now, true);
					answer[idx++] = now;
					flag = true;
					break;
				}
			}
		}
		
		return answer;
	}
}
